internal/jobworker/cgroup/cgroup_test.go

```diff
@@ -0,0 +1,233 @@
+		return
+	}
+}
+
+func TestCleanupWithCgroups(t *testing.T) {
+	if !isRoot() {
+		t.Skip("must be root to run")
+	}
+
+	dir := t.TempDir()
+	service, err := NewService(WithMountPath(dir))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	if _, err := service.CreateCgroup(); err != nil {
+		t.Error(err)
+	}
+
+	if err := service.Cleanup(); err != nil {
+		t.Error(err)
+		return
+	}
+
+	if _, err := os.Stat(service.path); !errors.Is(err, fs.ErrNotExist) {
+		t.Errorf("expected cgroup to not exist; path: %s, err: %v", service.path, err)
+		return
+	}
+}
+
+func TestCleanupWithPids(t *testing.T) {
+	if !isRoot() {
+		t.Skip("must be root to run")
+	}
+
+	dir := t.TempDir()
+	service, err := NewService(WithMountPath(dir))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	cgroup, err := service.CreateCgroup()
+	if err != nil {
+		t.Error(err)
+	}
+
+	cmd := exec.Command("sleep", "30")
+	if err := cmd.Start(); err != nil {
+		t.Error(err)
+	}
+
+	if err := service.PlaceInCgroup(*cgroup, cmd.Process.Pid); err != nil {
+		t.Error(err)
+	}
+
+	if err := service.Cleanup(); err != nil {
+		t.Error(err)
+	}
+
+	if _, err := os.Stat(service.path); !errors.Is(err, fs.ErrNotExist) {
+		t.Errorf("expected cgroup to not exist; path: %s, err: %v", service.path, err)
+		return
+	}
+}
+
+func TestCreateCgroup(t *testing.T) {
+	if !isRoot() {
+		t.Skip("must be root to run")
+	}
+
+	dir := t.TempDir()
+	service, err := NewService(WithMountPath(dir))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer func() {
+		if err := service.Cleanup(); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	tests := map[string]struct {
+		options []CgroupOption
+	}{
+		"no options":              {},
+		"w/ memory limit":         {options: []CgroupOption{WithMemory(1000000000)}},
+		"w/ cpu limit":            {options: []CgroupOption{WithCpus(1.5)}},
+		"w/ disk write bps limit": {options: []CgroupOption{WithDiskWriteBps(100000)}},
+		"w/ disk read bps limit":  {options: []CgroupOption{WithDiskReadBps(100000)}},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			cgroup, err := service.CreateCgroup(test.options...)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+
+			if _, err := os.Stat(cgroup.path); err != nil {
+				t.Errorf("expected cgroup to exist; path: %s", cgroup.path)
+				return
+			}
+		})
+	}
+}
+
+func TestPlaceInCgroup(t *testing.T) {
+	if !isRoot() {
+		t.Skip("must be root to run")
+	}
+
+	dir := t.TempDir()
+	service, err := NewService(WithMountPath(dir))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	defer func() {
+		if err := service.Cleanup(); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	cgroup, err := service.CreateCgroup()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	cmd := exec.Command("sleep", "30")
+	if err := cmd.Start(); err != nil {
+		t.Error(err)
+		return
+	}
+
+	if err := service.PlaceInCgroup(*cgroup, cmd.Process.Pid); err != nil {
+		t.Error(err)
+		return
+	}
+
+	pids, err := cgroup.readPids()
```

Comment:
nit: Let's not rely on private functions during tests.

Reply:
I agree, I was cutting corners. :smile: